File: 2021/test/day_09_test.rb
# frozen_string_literal: true

require "test_helper"
require "day_09"

class Day09Test < Minitest::Test
  def test_part1
    day = Day09.new

    assert_equal 15, day.part1(input("day_09_example_1"))
    assert_equal 494, day.part1(input("day_09"))
  end

  def test_part2
    day = Day09.new

    assert_equal 1134, day.part2(input("day_09_example_1"))
    assert_equal 1_048_128, day.part2(input("day_09"))
  end
end

File: 2025/test/day_09_test.exs
defmodule Day09Test do
  use ExUnit.Case

  import TestHelper

  test "part1" do
    assert Day09.part_1(input("day_09_example_1")) == 50
    assert Day09.part_1(input("day_09")) == 1594
  end

  # test "part2" do
  #   assert Day09.part_2(input("day_09_example_1")) == 40
  #   assert Day09.part_2(input("day_09")) == 15_650_261_281_478
  # end
end
